opensrp-pnc/src/main/java/org/smartregister/pnc/repository/PncPartialFormRepository.java

```diff
@@ -0,0 +1,96 @@
+import org.smartregister.repository.BaseRepository;
+
+import java.util.List;
+
+public class PncPartialFormRepository extends BaseRepository implements PncGenericDao<PncPartialForm> {
+
+    private static final String CREATE_TABLE_SQL = "CREATE TABLE " + PncDbConstants.Table.PNC_PARTIAL_FORM + "("
+            + PncDbConstants.Column.PncPartialForm.ID + " INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,"
+            + PncDbConstants.Column.PncPartialForm.BASE_ENTITY_ID + " VARCHAR NOT NULL, "
+            + PncDbConstants.Column.PncPartialForm.FORM + " TEXT NOT NULL, "
+            + PncDbConstants.Column.PncPartialForm.CREATED_AT + " INTEGER NOT NULL ," +
+            "UNIQUE(" + PncDbConstants.Column.PncPartialForm.BASE_ENTITY_ID + ") ON CONFLICT REPLACE)";
+
+    private static final String INDEX_BASE_ENTITY_ID = "CREATE INDEX " + PncDbConstants.Table.PNC_PARTIAL_FORM
+            + "_" + PncDbConstants.Column.PncPartialForm.BASE_ENTITY_ID + "_index ON " + PncDbConstants.Table.PNC_PARTIAL_FORM +
+            "(" + PncDbConstants.Column.PncPartialForm.BASE_ENTITY_ID + " COLLATE NOCASE);";
+
+    private String[] columns = new String[]{
+            PncDbConstants.Column.PncPartialForm.ID,
+            PncDbConstants.Column.PncPartialForm.BASE_ENTITY_ID,
+            PncDbConstants.Column.PncPartialForm.FORM,
+            PncDbConstants.Column.PncPartialForm.CREATED_AT};
+
+    public static void createTable(@NonNull SQLiteDatabase database) {
+        database.execSQL(CREATE_TABLE_SQL);
+        database.execSQL(INDEX_BASE_ENTITY_ID);
+    }
+
+    @Override
+    public boolean saveOrUpdate(@NonNull PncPartialForm pncPartialForm) {
+        ContentValues contentValues = new ContentValues();
+        contentValues.put(PncDbConstants.Column.PncPartialForm.BASE_ENTITY_ID, pncPartialForm.getBaseEntityId());
+        contentValues.put(PncDbConstants.Column.PncPartialForm.FORM, pncPartialForm.getForm());
+        contentValues.put(PncDbConstants.Column.PncPartialForm.CREATED_AT, pncPartialForm.getCreatedAt());
+        SQLiteDatabase sqLiteDatabase = getWritableDatabase();
+        long rows = sqLiteDatabase.insert(PncDbConstants.Table.PNC_PARTIAL_FORM, null, contentValues);
+        return rows != -1;
+    }
+
+    @Nullable
+    @Override
+    public PncPartialForm findOne(@NonNull PncPartialForm pncPartialForm) {
+        SQLiteDatabase sqLiteDatabase = getReadableDatabase();
+        Cursor cursor = sqLiteDatabase.query(PncDbConstants.Table.PNC_PARTIAL_FORM
+                , columns
+                , PncDbConstants.Column.PncPartialForm.BASE_ENTITY_ID + " = ? "
+                , new String[]{pncPartialForm.getBaseEntityId()}
+                , null
+                , null
+                , null);
+
+        if (cursor.getCount() == 0) {
+            return null;
+        }
+
+        PncPartialForm diagnosisAndTreatmentForm = null;
+        if (cursor.moveToNext()) {
+            diagnosisAndTreatmentForm = new PncPartialForm(
```

Comment:
i think there is a column called form type

Reply:
done